### DISCUSSION TITLE
add_requires是否支持自动链接

### DISCUSSION THREAD

**Discussion by leaf-potato** [xmake-io/xmake#3390] [Q&A]
在使用add_requires标签添加依赖包后
```lua
 add_requires("brpc 1.3.0")
 add_requires("toml++ 3.2.0")
 add_requires("gtest 1.12.1") 
```

如果想这些依赖包对所有target都生效的话 需要每个都`add_packages`一次：

``` lua
add_packages("brpc" "toml++" "gtest")
```

想问下 `add_requires`有配置项支持对所有的target都链接吗？这样可以不用再重复写一遍。或者`add_packages`支持*匹配吗？

**Comment by star-hengxing** [comment]
这个也可以用自定义 rule 解决，或者直接在全局配置域 add package

**Comment by waruqi** [comment]
add_requires 不处理 target links ，那是 add_packages 干的事情。。

你全局描述域设置 add_packages 不就好了，又不一定非得在 target 里面设置

**Reply by leaf-potato** [reply]
好吧 那只有挨个add_packages一次了

**Reply by waruqi** [reply]
我刚不是说了么，全局域设置，一次不就搞定了，为啥还要挨个设置？

**Reply by leaf-potato** [reply]
> 想问下 add_requires有配置项支持对所有的target都链接吗？这样可以不用再重复写一遍。或者add_packages支持*匹配吗？

在全局域设置可以对所有的target生效，这没问题。但问题在于当第三包很多时，想第三方包对所有target生效也需要在全局域挨个`add_packages`每个三方包。想问下这种情况下有没好的办法解决，比如`add_packages`是否支持像`add_files`的正则匹配

**Reply by waruqi** [reply]
至于这么懒么，包再多也没几个，搞个 list + for 不就完事了么

```
local list = {"zlib", "libpng", "openssl"}
for _, name in ipairs(list) do
    add_requires(name)
    add_packages(name)
end
target("A")
target("B")
```

或者自己搞个 function wrap 下，你在里面自己做模式匹配，自己搞。。

```
function add_package(name)
    add_requires(name)
    add_packages(name)
end
add_package("zlib")
add_package("openssl")
add_package("libpng")
target("A")
target("B")
```

再省也没得省了，总要知道装哪些包吧，我总不能凭空猜出来

**Reply by leaf-potato** [reply]
感觉解惑，这下学习到了，总得多问问，万一有更好的解决方法呢~
